### PR TITLE
Added support for reconfiguring the network

### PR DESCRIPTION
**Pull request by stewrutledge**
There isn't really any documentation updated in this one as this functionality seems implied in the module already.

Allows for the reconfiguration of network settings on virtual machines. Changing switch, switch type etc...

### CODE REVIEW

File: cloud/vsphere_guest.py [hunk 1]
```diff
@@ -67,7 +67,7 @@
     description:
      - Indicate desired state of the vm.
     default: present
-    choices: ['present', 'powered_on', 'absent', 'powered_off', 'restarted', 'reconfigured']
+    choices: ['present', 'powered_on', 'absent', 'powered_on', 'restarted', 'reconfigured']
```
Comment (jimi-c) [inline cloud/vsphere_guest.py]: This is a mistake, you've removed a valid option from the list of choices.

**Comment by stewrutledge** [comment]
Sorry, that typo existed in earlier versions of the module and must have followed along. It was a few commits back in devel and exists currently in the stable 1.7.

It's fixed now in this pull request.

**Comment by gregdek** [comment]
@stewrutledge thanks for submitting this pull request. Please accept our apologies for taking so long to review it.

Unfortunately, due to this delay, it appears that the underlying code has changed, and this PR needs to be rebased. If you believe that this PR still applies, please rebase and we will then put it into our review process as soon as possible.  Again, apologies for the delay and the extra work.

**Comment by gregdek** [comment]
@stewrutledge A friendly reminder: this pull request has been marked as needing your action, and it hasn't been updated in the last month. 

If you still believe that this PR applies, and you intend to address the issues with this PR, just let us know in the PR itself and we'll keep it open pending your changes.

If we receive no indication from you within a week, we will close this pull request.

**Comment by stewrutledge** [comment]
Sorry for the delay!

I'm not 100% sure what the status on reconfiguring network devices is, however I believe this functionality is still missing.

I'll investigate and open a new pull request (I don't think the actual modules code has changed so much that it will require a lot of work).

What is the best way to rebase this all the way up to the current ansible version? At least one major release, and quite a few changes have come since it was originally opened.

**Comment by gregdek** [comment]
Hi @stewrutledge -- no worries. Thanks for getting back to us. :)

I'd recommend that you stash the file, reset your git repo so that it's in sync with upstream, copy the file back in (and resolve any conflicts that have emerged in the meantime) and then open a new pull request against devel. That should take care of the conflicts. 

It's probably also best to close this PR in expectation of a shiny new one, so I'll do that now.   When you do create the new one, refer to this one in the comments. 

Thanks again!

**Pull request closed without merging**